transaction/simple.go

```diff
@@ -2,29 +2,35 @@ package transaction
 
 import (
 	"github.com/gluster/glusterd2/context"
+
+	"github.com/pborman/uuid"
 )
 
 // SimpleTxn is transaction with fixed stage, commit and store steps
 type SimpleTxn struct {
```

Comment:
The SimpleTxn resembles the existing transaction mechanism in GD-1, which is a two-phase commit. It brings in a much needed rollback which isn't present GD-1.

The stage step is the commit-request phase, and the commit step is the commit-phase. In addition to these We have lock and unlock steps to handle concurrent actions on a resource.

In the stage step, the leader (or co-ordinator) requests each follower (or cohort) to validate that they can perform the commit. The commit is done once everyone replies with a yes. If the commit fails on anyone, any changes done are rolled back.

We do have a problem here, as is common in a 2-phase commit, that we don't have a timeout. This is something we'll need to address.